### ISSUE TITLE
pm2 and nginx to deploy

### ISSUE THREAD

**Issue by serbol** [Gottwik/Enduro#123]
Is it possible to deploy Enduro web page using pm2 and nginx to deploy?
How it can be done?

**Comment by JoeSquishy** [comment]
Yeah, I'm fairly sure I got it working yesterday on digital ocean.

I followed this guide for the nginx: https://www.howtoforge.com/tutorial/how-to-deploy-nodejs-applications-with-pm2-and-nginx-on-ubuntu/#step-install-and-configure-nginx-as-a-reverse-proxy

I started pm2 with path "node_modules/.bin/enduro" -- start (roughly)

Sorry if its not much help - fairly new to this

**Comment by jemelyah** [comment]
Here is my ecosystem.config.js for pm2 in the repo top-folder. For this to work You should have PM2 installed locally and remotely and also nginx up and running remotely. You can try it and fine tune to Your needs. Pre-deploy setup with 'pm2 deploy dev setup' and deploy with 'pm2 deploy dev' then.

module.exports = {
  /**
   * Application configuration section
   * http://pm2.keymetrics.io/docs/usage/application-declaration/
   */
  apps : [

    // Staging application
    {
      name      : 'Staging',
      script    : 'enduro',
      args: 'start',
      cwd: '/var/www/application/development/current',
      instances : 0,
      exec_mode : "cluster",
      env: {
        COMMON_VARIABLE: 'true',
        PORT: '4444'
      },
      env_development : {
        NODE_ENV: 'dev'
      }
    },
    // Production application
    {
      name      : 'application',
      script    : 'enduro',
      args: 'start',
      cwd: '/var/www/application/production/current',
      instances : 0,
      exec_mode : "cluster",
      env: {
        COMMON_VARIABLE: 'true',
        PORT: '4445'
      },
      env_production : {
        NODE_ENV: 'production'
      }
    }
  ],

  /**
   * Deployment section
   * http://pm2.keymetrics.io/docs/usage/deployment/
   */
  deploy : {
    production : {
      key: '~/.ssh/id',
      user : 'root',
      host : 'prod_ip_adress',
      ref  : 'origin/master',
      repo : 'git@bitbucket.org:repo.git',
      path : '/var/www/application/production',
      
      'post-deploy' : 'rm -f /etc/nginx/sites-available/applicationProxy && rm -f /etc/nginx/sites-enabled/applicationProxy && npm install && pm2 reload ecosystem.config.js --only application --env production && ln -s /var/www/application/production/source/applicationProxy /etc/nginx/sites-available/applicationProxy && ln -s /var/www/application/production/source/applicationProxy /etc/nginx/sites-enabled/applicationProxy && service nginx restart',
      'ssh_options' : ["StrictHostKeyChecking=no", "PasswordAuthentication=no"]
    },
    dev : {
      key: '~/.ssh/id',
      user : 'root',
      host : 'dev_ip_adress',
      ref  : 'origin/dev',
      repo : 'git@bitbucket.org:repo.git',
      path : '/var/www/application/development',
      'post-deploy' : 'rm -f /etc/nginx/sites-available/applicationStaging && rm -f /etc/nginx/sites-enabled/applicationStaging && npm install && pm2 reload ecosystem.config.js --only Staging --env dev && ln -s /var/www/application/development/source/applicationStaging /etc/nginx/sites-available/applicationStaging && ln -s /var/www/application/development/source/applicationStaging /etc/nginx/sites-enabled/applicationStaging && service nginx restart',
      'ssh_options' : ["StrictHostKeyChecking=no", "PasswordAuthentication=no"],
      env  : {
        NODE_ENV: 'dev'
      }
    }
  }
};

So, the applicationProxy and applicationStaging are nginx configs, the applicationStaging looks like:

upstream Staging {
    server ip:4444;
    keepalive 64;
}

server {
    listen 80;
    server_name Staging;
    root /var/www/application/development/current/_generated/;

    location / {
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header Host $http_host;
        proxy_set_header X-NginX-Proxy true;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_max_temp_file_size 0;
        proxy_pass http://Staging/;
        proxy_redirect off;
        proxy_read_timeout 240s;
    }
}

**Comment by jemelyah** [comment]
So, all those file are in the root of my repo.
